preview/MsixCore/msixmgr/ApplyACLsProvider.cpp

```diff
@@ -0,0 +1,52 @@
+#include "MSIXWindows.hpp"
+#include "ApplyACLsProvider.hpp"
+#include "msixmgrLogger.hpp"
+#include "..\msixmgrLib\GeneralUtil.hpp"
+#include <string>
+#include <iostream>
+#include <vector>
+#include <TraceLoggingProvider.h>
+#include "InstallUI.hpp"
+
+using namespace MsixCoreLib;
+using namespace std;
+
+namespace MsixCoreLib
+{
+    HRESULT ApplyACLsProvider::ApplyACLs(
+        _In_ std::vector<std::wstring> packageFolders)
+    {
+        auto autoFreeLibrary = [](HMODULE* module)
+        {
+            if (module != nullptr)
+            {
+                FreeLibrary(*module);
+            }
+        };
+
+        std::unique_ptr<HMODULE, decltype(autoFreeLibrary)>
+            applyACLsDll(nullptr, autoFreeLibrary);
+
+        HMODULE applyACLsLocal = LoadLibrary(L"applyacls.dll");
+        applyACLsDll.reset(&applyACLsLocal);
+
+        if (applyACLsDll == nullptr)
+        {
+            std::wcout << "Failed to load applyacls.dll. Please confirm the dll is next to this exe" << std::endl;
+        }
+
+        typedef HRESULT(STDMETHODCALLTYPE *APPLYACLSTOPACKAGEFOLDER)(PCWSTR folderPath);
+
+        APPLYACLSTOPACKAGEFOLDER ApplyACLsToPackageFolder =
+            reinterpret_cast<APPLYACLSTOPACKAGEFOLDER>
+            (GetProcAddress(*applyACLsDll, "ApplyACLsToPackageFolder"));
+
+        for (auto folder : packageFolders)
+        {
+            std::wcout << folder << std::endl;

```

Comment:
> std::wcout << folder << std::endl; [](start = 11, length = 35)

still here.

Reply:
you haven't addressed this. is it intended for the user to see? you should be able to #include "msixmgrLogger.hpp"  and use TraceLoggingWrite (search in InstallUI.cpp for example)

---
In reply to: [314869606](https://github.com/microsoft/msix-packaging/pull/201#discussion_r314869606) [](ancestors = 314869606)

Reply:
I've still working on validating other parts of the code and was planning to circle back to this later. That being said, do we really want to be logging this to the trace provider? This could literally be hundreds of folder names that are getting logged. This was originally added just so I knew that the correct folders were getting processed

Reply:
oh ok. i  didn't know whether you're waiting on me to review or whether you were doing something else. You definitely don't want it to be cout to user. You can delete if you think there's little value; if there's any value in helping debug in the future, then logging to the trace provider is fine. Trace provider is intended to be noisy--nobody is going to look at it unless something goes wrong.

---
In reply to: [314919298](https://github.com/microsoft/msix-packaging/pull/201#discussion_r314919298) [](ancestors = 314919298)